### PR TITLE
fix: auto verify google sso emails

### PR DESCRIPTION
**Pull request by dguyen**
## Description

Currently logging in via Google SSO will still require you to verify your email. This shouldn't happen since going through Google SSO implies you have access to that email.

## Changes Made

This change resolves that by setting the `emailVerified` attribute in the database when you access the application via Google SSO.

This also updates the identity provider for users who log in through Google to have it correctly set as `GOOGLE`.

## Testing Performed

- Logged in via SSO and no email verification was requested
- Logged in via non SSO and email verification was requested

## Checklist

- [X] I have tested these changes locally and they work as expected.
- [X] I have followed the project's coding style guidelines.

### CODE REVIEW

File: packages/lib/next-auth/auth-options.ts [hunk 1]
```diff
@@ -138,6 +138,23 @@ export const NEXT_AUTH_OPTIONS: AuthOptions = {
         merged.emailVerified = user.emailVerified?.toISOString() ?? null;
       }
 
+      if (
+        (trigger === 'signIn' || trigger === 'signUp') &&
+        merged.emailVerified === null &&
+        account?.provider === 'google'
```
Comment (Mythie) [inline packages/lib/next-auth/auth-options.ts:144]: Does this work? If it does that's amazing ❤️

Comment (dguyen) [reply packages/lib/next-auth/auth-options.ts:144]: Pho sho 🙏

Comment (dguyen) [reply packages/lib/next-auth/auth-options.ts:144]: I just realized that we're not using setting `identityProvider` on the user table to `GOOGLE` 

Should I just slide that in here? We need it set correctly for this PR: #709

**Comment by dguyen** [comment]
Applied an update so that whenever the user signs in or signs up via Google, we will update the user identity provider to be `GOOGLE`.

We can't do this conditionally, so this will happen every time during sign in or sign up via Google SSO.

File: packages/lib/next-auth/auth-options.ts [hunk 2]
```diff
@@ -138,6 +139,22 @@ export const NEXT_AUTH_OPTIONS: AuthOptions = {
         merged.emailVerified = user.emailVerified?.toISOString() ?? null;
       }
 
+      if ((trigger === 'signIn' || trigger === 'signUp') && account?.provider === 'google') {
```
Comment (dguyen) [inline packages/lib/next-auth/auth-options.ts:142]: It's not optimal to run this every sign up or sign in via Google, but we don't have access to `identityProvider` to see if it's correctly set.

Comment (Mythie) [reply packages/lib/next-auth/auth-options.ts:142]: We could easily skip out if we know the user is already verified no?

Comment (dguyen) [reply packages/lib/next-auth/auth-options.ts:142]: It would, but we don't know if the provider is correctly set on the user table

Comment (Mythie) [reply packages/lib/next-auth/auth-options.ts:142]: Slap it in the token while we're here, this saves tripping to the db all the time 😄

Comment (Mythie) [reply packages/lib/next-auth/auth-options.ts:142]: Actually don't worry about it, we trip enough as is and this is only on signIn or signUp

**Review by Mythie — APPROVED** [review]

**Pull request merged**